apps/frontend/src/pages/tap.tsx

```diff
@@ -139,17 +146,33 @@ const TapPage: React.FC = () => {
             // Save tap to populate registration flow data
             logClientEvent("tap-chip-not-registered-not-logged-in", {});
             await storage.saveTapInfo({ tapParams, tapResponse: response });
+
             router.push("/register");
             return;
-          } else {
-            // For now, we do not allow users to register chips after account creation
-            // TODO: Allow users to register chips after account creation
-            logClientEvent("tap-chip-not-registered-logged-in", {});
-            toast.error(
-              "This chip is not registered yet. Please tell your friend to register it first!"
-            );
+          } else if (response.isLocationChip !== true && session && user?.userData) {
+            // User logged in and unregistered chip is not locationChip, allow user to bind new chip to profile
+            logClientEvent("tap-chip-logged-in-bind-new-chip", {});
+
+            const userData = JsonSchema.parse(user.userData);
```

Comment:
this should be updated to use a function that selects the appropriate data for users, see comment below